src/utilities/include/optional_assert.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 The Manifold Authors, Jared Hoberock and Nathan Bell of
+// NVIDIA Research
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "structs.h"
+
+#ifdef MANIFOLD_DEBUG
+#include <iostream>
+#include <sstream>
+#include <string>
+
+template <typename Ex>
+void Assert(bool condition, const char* file, int line, const std::string& cond,
+            const std::string& msg) {
+  if (!condition) {
+    std::ostringstream output;
+    output << "Error in file: " << file << " (" << line << "): \'" << cond
+           << "\' is false: " << msg;
+    throw Ex(output.str());
+  }
+}
+#define ASSERT(condition, EX, msg) \
+  Assert<EX>(condition, __FILE__, __LINE__, #condition, msg);
+#else
+#define ASSERT(condition, EX, msg)
```

Comment:
@pca006132 Replying to you here to give context. When `MANIFOLD_DEBUG` is not defined, the `ASSERT` macro (which replaced `ALWAYS_ASSERT`) should compile out completely so there are no `throw` statements left. However, it now seg faults, so I think my macro is expanding in a way I don't expect.

Reply:
Nevermind, I found the problem. I had a side-effect inside one of my `ASSERT` calls :roll_eyes: